cmd/lifecycle/creator.go

```diff
@@ -144,7 +144,7 @@ func (c *createCmd) Exec() error {
 			lifecycle.NewConfigHandler(),
 			dirStore,
 		)
-		detector, err := detectorFactory.NewDetector(c.AppDir, c.BuildConfigDir, c.OrderPath, c.PlatformDir, cmd.DefaultLogger)
+		detector, err := detectorFactory.NewDetector(c.AppDir, c.BuildConfigDir, c.OrderPath, c.PlatformDir, c.AnalyzedPath, cmd.DefaultLogger)
```

Comment:
just for clarity --  "alphabetized but the logger is still last"?  or "alphabetized including the logger"?  I'm not immediately spotting another place where we do this. 

what if we make ... a new struct ... for making our other struct?  😬  e.g. 
```
struct DetectorInputs {  // i'm probably pretending Go uses C syntax here but let's be honest, if there was a sword stuck in a stone C would be the only language capable of pulling it out to become the one true compiler of England
  AppDir string
  BuildConfigDir string
  OrderPath string
  PlatformDir string
  AnalyzedPath string
  Logger PaulBunyan // whatever the right type is, idk :) 
}
```
then we could pass one of those structs in,  and basically it's just a hack to make Go give us keyword arguments , so the function call would look like:

`detectorFactory.NewDetector(DetectorInputs{AppDir: c.Appdir, BuildConfigDir: c.BuildConfigDir, ...})` 

OR honestly,  the only time we _don't_ already have a struct that we're just copying all the info out of is the unit tests,  so we could even make some convenience shims:
`detectorFactory.NewFromCreator(creator)` and `detectorFactory.NewFromDetectCmd(d)` .... oh or maybe those would be a little roundabout to avoid import cycles, but still.  overall I vaguely like the idea of using structs to replace long args lists (i.e. i miss python's kwargs) and/or having specialized functions to construct the new object from the object that already has all the data we want (i.e. i miss C++'s function overloading),  but as I'm calling out I'm also definitely guilty of wanting Go to be more like its cousins and less like itself,  so maybe there's some good arguments against such thoughts.

Reply:
ok thx --  i think using `platform.LifecycleInputs` wouldn't give us a way to indicate clearly which fields are optional in the larger struct.   I also think since this only gets called from a 2 places (plus unit tests) it feels like over-engineering to do any of the heavier things i outlined above.   So that leaves alphabetizing...  but fwiw i had to sing the alphabet song to figure out which comes first: "analyzedPath" or "appDir",  so apologies if i still got it wrong  :)